### PR TITLE
visualization PR

### PR DESCRIPTION
**Pull request by lyie28**
-> Facet class, plus original curve/scatter + histplot corrected + setup.py updated

### CODE REVIEW

File: client/src/bastionlab/polars/remote_polars.py [hunk 1]
```diff
@@ -289,64 +291,110 @@ def join_asof(
...
-            .apply_udf([col_x], model)
-            .groupby(pl.col(col_x))
-            .agg(pl.col(col_y).count())
-            .sort(col_x)
-            .collect()
-            .fetch()
-            .to_pandas()
-        )
-        sns.barplot(x=df[col_x], y=df[col_y], **kwargs)
+    def histplot(self: LDF, **kwargs):
```
Comment (dhalf) [inline client/src/bastionlab/polars/remote_polars.py:294]: I would use explicit parameters for everything you use in the body and don't just forward to seaborn. That way, they will appear in the signature and be typed and the default value will also be explicit.

**Comment by dhalf** [comment]
Also, is there a notebook that shows these features?
I would do that in the visualization tutorial

**Comment by lyie28** [comment]
1) Visualization tuto is done.
2) histplot() specifies x, y and bins arguments
3) Error handling if cols don't exist
4) 2d facet grid implemented as previously only worked in rows and cols specified, not just one!

File: client/src/bastionlab/polars/remote_polars.py [hunk 2]
```diff
@@ -313,64 +315,126 @@ def join_asof(
             ),
         )
 
-    def histplot(self: LDF, col_x: str, col_y: str, bins: int = 10, **kwargs):
+    def histplot(
```
Comment (JoFrost) [inline client/src/bastionlab/polars/remote_polars.py:318]: Tutorials and all is good, but personally, I would really like to see a docstring associated to the histplot method. You can use as an example what we did on BlindAI: https://github.com/mithril-security/blindai/blob/075929dd5e030983a0c153e1c6052297eca6f53a/client/blindai/client.py#L602

Comment (lyie28) [reply client/src/bastionlab/polars/remote_polars.py:318]: -> Done docstring for all visualize/Facet methods ✅ 
-> Also implemented kwargs for Facet class properly as realised I was not forwarding kwargs to subplots() in Facet functions

File: client/src/bastionlab/polars/remote_polars.py [hunk 3]
```diff
@@ -313,64 +315,161 @@ def join_asof(
...
+        self: LDF, x: str = "count", y: str = "count", bins: int = 10, **kwargs
+    ):
+        """Histplot plots a univariate histogram, where one x or y axes is provided or a bivariate histogram, where both x and y axes values are supplied.
```
Comment (JoFrost) [inline client/src/bastionlab/polars/remote_polars.py:321]: So, if I understand correctly this doctoring, it is not raising any exception, correct?
Does Seaborn raise exceptions?

File: client/src/bastionlab/polars/remote_polars.py [hunk 4]
```diff
@@ -313,64 +315,161 @@ def join_asof(
...
-        )
-        sns.barplot(x=df[col_x], y=df[col_y], **kwargs)
+
+        # if we have only X or Y
+        if col_x == "count" or col_y == "count":
+            q_x = pl.col(col_x) if col_x != "count" else pl.col(col_y)
+            q_y = pl.count()
+
+            if not col_x in self.columns and not col_y in self.columns:
+                print("Error: column not found in dataframe")
```
Comment (JoFrost) [inline client/src/bastionlab/polars/remote_polars.py:348]: Why not raising an exception here? This way the user can catch it and have a clear idea of what went wrong.

File: client/src/bastionlab/polars/remote_polars.py [hunk 5]
```diff
@@ -313,64 +315,161 @@ def join_asof(
...
+            x (str): The name of column to be used for x axes.
+            y (str): The name of column to be used for y axes.
+            order (int): If order is greater than 1, Seaborn's regplot uses numpy.polyfit to estimate a polynomial regression. Default value is 3.
+            ci: Size of the confidence interval for the regression estimate. Default value is None.
+            scatter (bool): Option to display a scatterplot of data with the underlying observations. Default value is False.
+            **kwargs: Other keyword arguments that will be passed to Seaborn's regplot function.
+        """
+        for col in [x, y]:
+            if not col in self.columns:
+                print("Error: ", col, " does not exist in dataframe")
```
Comment (JoFrost) [inline client/src/bastionlab/polars/remote_polars.py:428]: Same here and on the other methods, does the method can raise exceptions, even the ones that are triggered by Seaborn?

Comment (JoFrost) [inline client/src/bastionlab/polars/remote_polars.py:428]: Also, like on the others error print, I think an exception is more suited to show an error.

Comment (lyie28) [reply client/src/bastionlab/polars/remote_polars.py]: Implemented an exception for these now :+1:

**Review by JoFrost — APPROVED** [review]
Looks good to me. Thanks.

**Review by dhalf — CHANGES_REQUESTED** [review]
Can we use seaborn.lineplot instead of seaborn.regplot?
This has two benefits:
- lineplot behaves like the very common plt.plot, which means we can have several curves on the same plot with "hue"
- seaborn does not try to fit a model like with regplot so the data is genuinely displayed
Overall this sounds like a better default than regplot.
And we could of course support regplot in the future.

**Pull request merged**